File: Yixuan/18_01_17/drop_eggs.java
public class Solution {
    /**
     * @param n an integer
     * @return an integer
     * @link : http://www.lintcode.com/en/problem/drop-eggs/
     * @author: Egbert Li
     */
    public int dropEggs(int n) {
        long ans = 0;
        int x = 0; // asume start from Xth flooer
        while (ans < n) {
            x += 1;
            ans += x;
        }
        return x;
    }
}

File: Yixuan/18_01_17/drop_eggs.py
class Solution:
    # @param {int} n an integer
    # @return {int} an integer
    # @link: http://www.lintcode.com/en/problem/drop-eggs/
    # @author : Egbert Li
    def dropEggs(self, n):
        ans = 0
        x = 0
        while ans < n:
            x += 1
            ans += x
        return x
